### PR TITLE
chore: add angular 16 test infrastructure 

### PR DESCRIPTION
**Pull request by sean-perkins**
## What is the current behavior?

Issue URL: Internal

## What is the new behavior?

- Adds Angular 16 test app
- Introduces version test to validate which major version of Angular is being tested

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: angular/test/apps/ng16/e2e/src/angular-version.spec.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+it("should be on Angular 16", () => {
```
Comment (sean-perkins) [inline angular/test/apps/ng16/e2e/src/angular-version.spec.ts:1]: Need to add the Angular version test for both ng14 and ng15 apps.

File: .github/workflows/actions/test-angular-e2e/action.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ runs:
       shell: bash
       working-directory: ./angular/test
     - name: Install Dependencies
-      run: npm install
+      run: npm install --legacy-peer-deps
```
Comment (sean-perkins) [inline .github/workflows/actions/test-angular-e2e/action.yml:32]: Need to create a ticket to track swapping this once stable versions of v16 dependencies land. Alternatively we could continue to allow legacy peer dependencies for the test apps. The only advantage I see to not allowing it currently, is if some package we are using is incompatible with the version of Angular we are testing with. Our tests _should_ catch this scenario however. Unfortunately it seems any time we try to add the next major version test app before official release, we have to change this.

File: angular/test/apps/ng16/src/app/app.component.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<ion-app>
```
Comment (sean-perkins) [inline angular/test/apps/ng16/src/app/app.component.html:1]: With ng14 being the minimal supported version and no longer having the `[environmentInjector]` optional syntax, I can do away with the duplicate `AppComponent` for each individual test app and move this to the base app.

File: angular/test/apps/ng16/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  "compileOnSave": false,
+  "compilerOptions": {
+    "outDir": "./dist/out-tsc",
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "noImplicitOverride": true,
+    "noPropertyAccessFromIndexSignature": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+    "sourceMap": false,
```
Comment (sean-perkins) [inline angular/test/apps/ng16/tsconfig.json:11]: I had to disable source maps for Cypress to work (would throw an exception when opening cypress). This clearly isn't an issue in CI for the other test apps. I should re-enable and see if it is just a local Cypress issue (perhaps and outdated dmg).

File: .github/workflows/actions/test-angular-e2e/action.yml (hunk 2 above)
Comment (sean-perkins) [reply .github/workflows/actions/test-angular-e2e/action.yml:32]: Created FW-4092 to track.

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**